File: frontend/src/app/model/product.ts
export class Product {

  id: number;
  name: string;
  price: number;
  description: string;
  image: string;
  date_created: Date;
  date_updated: Date;


  constructor(id: number, name: string, price: number, description: string, image: string, date_created: Date, date_updated: Date) {
    this.id = id;
    this.name = name;
    this.price = price;
    this.description = description;
    this.image = image;
    this.date_created = date_created;
    this.date_updated = date_updated;
  }
}

File: backend/src/main/java/com/amelectronic/market/model/Product.java
package com.amelectronic.market.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.Lob;
import javax.persistence.ManyToOne;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Product extends CategoryProduct {

    private int price;
    private String image;

    @Lob
    private String description;

    @JsonIgnore
    @ManyToOne
    @JoinColumn(name = "id_category")
    private Category category;
}
